### ISSUE TITLE
Eigen matrix problem in MinDistanceConstraint

### ISSUE THREAD

**Issue by hongkai-dai** [RobotLocomotion/drake#627]
I just tried to compile Drake in debug mode, and then run testIK. It fails in checking the MinDistanceConstraint in line 314 of `testIK`. The error message is as follows:
```
Check all-to-all closest distance constraint
Elapsed time is 2.002276 seconds.
Check minimum-distance constraint
MATLAB: /home/hongkai/drake-distro/build/include/eigen3/Eigen/src/Core/CommaInitializer.h:95: Eigen::CommaInitializer<MatrixType>::~CommaInitializer() [with XprType = Eigen::Matrix<double, -0x00000000000000001, -0x00000000000000001>]: Assertion `(m_row+m_currentBlockRows) == m_xpr.rows() && m_col == m_xpr.cols() && "Too few coefficients passed to comma initializer (operator<<)"' failed.
```
@avalenzu , I think we have a problem in in collision code when we use Eigen.

**Comment by RussTedrake** [comment]
@avalenzu - could you make this one a priority since it seems important and shouldn't be hard to fix (if it hasn't been fixed already)?

**Comment by avalenzu** [comment]
Will do

**Comment by avalenzu** [comment]
Sorry, forgot that this was already resolved by 62591139. 

OK to close.
